profiles/views.py

```diff
@@ -50,13 +49,10 @@ def test_func(self):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context["discord_role_membership"] = RoleMembership.objects.filter(
-            guild_id=settings.DISCORD_GUILD_ID, user_id=self.get_object().id
-        ).order_by(
-            "-role__position"
-        ).exclude(
-            role__name="@everyone"
-        )
+        try:
+            context['socialaccount'] = SocialAccount.objects.get(user=self.get_object())
+        except SocialAccount.DoesNotExist:
+            context['socialaccount'] = None
```

Comment:
Don't `get()` methods usually have a way to specify a default? Then you can just have a single line.